### ISSUE TITLE
Receive ValueError when saving in GeoPackage format

### ISSUE THREAD

**Issue by andrewalseth** [gboeing/osmnx#516]
#### Problem description (what did you do, what did you expect to happen, and what actually happened)
I downloaded a graph from place with a filter, reprojected the graph to EPSG:3857, consolidated the intersections and attempted to save the resulting output to a geopackage format. Received ValueError each time, as below.

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-153-6be1e98cdb9d> in <module>
----> 1 ox.io.save_graph_geopackage(G_test, 'data/Kirkwood_test.gpkg')

~/opt/anaconda3/envs/ox/lib/python3.8/site-packages/osmnx/io.py in save_graph_geopackage(G, filepath, encoding)
     51 
     52     # save the nodes and edges as GeoPackage layers
---> 53     gdf_nodes.to_file(filepath, layer="nodes", driver="GPKG", encoding=encoding)
     54     gdf_edges.to_file(filepath, layer="edges", driver="GPKG", encoding=encoding)
     55     utils.log(f'Saved graph as GeoPackage at "{filepath}"')

~/opt/anaconda3/envs/ox/lib/python3.8/site-packages/geopandas/geodataframe.py in to_file(self, filename, driver, schema, index, **kwargs)
    531         from geopandas.io.file import to_file
    532 
--> 533         to_file(self, filename, driver, schema, index, **kwargs)
    534 
    535     def to_crs(self, crs=None, epsg=None, inplace=False):

~/opt/anaconda3/envs/ox/lib/python3.8/site-packages/geopandas/io/file.py in to_file(df, filename, driver, schema, index, **kwargs)
    191             filename, "w", driver=driver, crs_wkt=crs_wkt, schema=schema, **kwargs
    192         ) as colxn:
--> 193             colxn.writerecords(df.iterfeatures())
    194 
    195 

~/opt/anaconda3/envs/ox/lib/python3.8/site-packages/fiona/collection.py in writerecords(self, records)
    340         if self.mode not in ('a', 'w'):
    341             raise IOError("collection not open for writing")
--> 342         self.session.writerecs(records, self)
    343         self._len = self.session.get_length()
    344         self._bounds = self.session.get_extent()

fiona/ogrext.pyx in fiona.ogrext.WritingSession.writerecs()

fiona/ogrext.pyx in fiona.ogrext.OGRFeatureBuilder.build()

ValueError: Invalid field type <class 'numpy.int64'>


```

Saving as GraphML or shapefile format does not create a ValueError, however.

#### What operating system, Python version, and OSMnx version are you using?
OSX 10.15.4, Python 3.8, OSMnx 0.14.1

#### Complete list of your environment's packages and their versions (for example, run `conda list` or `pip list` then paste the output between the two "details" tags below)

<details>
# Paste the output of your Python packages and their versions here, between these two "details" tags
appnope                   0.1.0           py38h32f6830_1001    conda-forge
attrs                     19.3.0                     py_0    conda-forge
backcall                  0.1.0                      py_0    conda-forge
bleach                    3.1.5              pyh9f0ad1d_0    conda-forge
boost-cpp                 1.72.0               hdf9ef73_0    conda-forge
branca                    0.4.1                      py_0    conda-forge
brotlipy                  0.7.0           py38h64e0658_1000    conda-forge
bzip2                     1.0.8                h0b31af3_2    conda-forge
ca-certificates           2020.4.5.2           hecda079_0    conda-forge
cairo                     1.16.0            hec6a9b0_1003    conda-forge
certifi                   2020.4.5.2       py38h32f6830_0    conda-forge
cffi                      1.14.0           py38h356ff06_0    conda-forge
cfitsio                   3.470                hddc9c06_5    conda-forge
chardet                   3.0.4           py38h32f6830_1006    conda-forge
click                     7.1.2              pyh9f0ad1d_0    conda-forge
click-plugins             1.1.1                      py_0    conda-forge
cligj                     0.5.0                      py_0    conda-forge
coverage                  5.1              py38h64e0658_0    conda-forge
cryptography              2.9.2            py38hb761dab_0    conda-forge
curl                      7.69.1               h2d98d24_0    conda-forge
cycler                    0.10.0                     py_2    conda-forge
dbus                      1.13.6               h2f22bb5_0    conda-forge
decorator                 4.4.2                      py_0    conda-forge
defusedxml                0.6.0                      py_0    conda-forge
descartes                 1.1.0                      py_4    conda-forge
entrypoints               0.3             py38h32f6830_1001    conda-forge
exiv2                     0.27.1               ha4feea6_0    conda-forge
expat                     2.2.9                h4a8c4bd_2    conda-forge
fiona                     1.8.13           py38he71f6a4_0    conda-forge
folium                    0.11.0                     py_0    conda-forge
fontconfig                2.13.1            h6b1039f_1001    conda-forge
freetype                  2.10.2               h8da9a1a_0    conda-forge
freexl                    1.0.5             h1de35cc_1002    conda-forge
future                    0.18.2           py38h32f6830_1    conda-forge
gdal                      3.0.4           py38h9f7df5a_10    conda-forge
geopandas                 0.7.0                      py_1    conda-forge
geos                      3.8.1                h4a8c4bd_0    conda-forge
geotiff                   1.6.0                hd8796ba_0    conda-forge
gettext                   0.19.8.1          h46ab8bc_1002    conda-forge
giflib                    5.2.1                h0b31af3_2    conda-forge
glib                      2.64.3               h577aef8_0    conda-forge
gsl                       2.6                  ha2d443c_0    conda-forge
hdf4                      4.2.13            h84186c3_1003    conda-forge
hdf5                      1.10.6          nompi_h3e39495_100    conda-forge
httplib2                  0.18.1             pyh9f0ad1d_0    conda-forge
icu                       64.2                 h6de7cb9_1    conda-forge
idna                      2.9                        py_1    conda-forge
importlib-metadata        1.6.1            py38h32f6830_0    conda-forge
importlib_metadata        1.6.1                         0    conda-forge
ipykernel                 5.3.0            py38h23f93f0_0    conda-forge
ipython                   7.15.0           py38h32f6830_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
jedi                      0.17.0           py38h32f6830_0    conda-forge
jinja2                    2.11.2             pyh9f0ad1d_0    conda-forge
joblib                    0.15.1                     py_0    conda-forge
jpeg                      9d                   h0b31af3_0    conda-forge
json-c                    0.13.1            h575e443_1002    conda-forge
json5                     0.9.0                      py_0    conda-forge
jsonschema                3.2.0            py38h32f6830_1    conda-forge
jupyter_client            6.1.3                      py_0    conda-forge
jupyter_core              4.6.3            py38h32f6830_1    conda-forge
jupyterlab                2.1.4                      py_0    conda-forge
jupyterlab_server         1.1.5                      py_0    conda-forge
jupyterthemes             0.20.0                     py_1    conda-forge
kealib                    1.4.13               h40102fb_1    conda-forge
kiwisolver                1.2.0            py38ha0d09dd_0    conda-forge
krb5                      1.17.1               h1752a42_0    conda-forge
lesscpy                   0.13.0                     py_1    conda-forge
libblas                   3.8.0               16_openblas    conda-forge
libcblas                  3.8.0               16_openblas    conda-forge
libclang                  9.0.1           default_hf57f61e_0    conda-forge
libcurl                   7.69.1               hc0b9707_0    conda-forge
libcxx                    10.0.0               h1af66ff_2    conda-forge
libdap4                   3.20.6               h8c15375_0    conda-forge
libedit                   3.1.20191231         hed1e85f_0    conda-forge
libffi                    3.2.1             h4a8c4bd_1007    conda-forge
libgdal                   3.0.4               h242383b_10    conda-forge
libgfortran               4.0.0                         2    conda-forge
libiconv                  1.15              h0b31af3_1006    conda-forge
libkml                    1.3.0             h169b8f9_1011    conda-forge
liblapack                 3.8.0               16_openblas    conda-forge
libllvm9                  9.0.1                h7475705_1    conda-forge
libnetcdf                 4.7.4           nompi_ha11d67f_104    conda-forge
libopenblas               0.3.9                h3d69b6c_0    conda-forge
libpng                    1.6.37               hbbe82c9_1    conda-forge
libpq                     12.2                 h489d428_1    conda-forge
libsodium                 1.0.17               h01d97ff_0    conda-forge
libspatialindex           1.9.3                h4a8c4bd_3    conda-forge
libspatialite             4.3.0a            h658e6c1_1038    conda-forge
libssh2                   1.9.0                h39bdce6_2    conda-forge
libtasn1                  4.14                 h01d97ff_0    conda-forge
libtiff                   4.1.0                h2ae36a8_6    conda-forge
libwebp-base              1.1.0                h0b31af3_3    conda-forge
libxml2                   2.9.10               h53d96d6_0    conda-forge
libzip                    1.5.1             h22952d1_1005    conda-forge
llvm-openmp               10.0.0               h28b9765_0    conda-forge
lz4-c                     1.9.2                h4a8c4bd_1    conda-forge
markupsafe                1.1.1            py38h64e0658_1    conda-forge
matplotlib-base           3.2.1            py38h1300a51_0    conda-forge
mistune                   0.8.4           py38h64e0658_1001    conda-forge
mock                      4.0.2            py38h32f6830_0    conda-forge
munch                     2.5.0                      py_0    conda-forge
nbconvert                 5.6.1            py38h32f6830_1    conda-forge
nbformat                  5.0.6                      py_0    conda-forge
ncurses                   6.1               h0a44026_1002    conda-forge
networkx                  2.4                        py_1    conda-forge
nose2                     0.9.2                      py_0    conda-forge
notebook                  6.0.3            py38h32f6830_0    conda-forge
nspr                      4.20              h0a44026_1000    conda-forge
nss                       3.47                 hc0980d9_0    conda-forge
numpy                     1.18.5           py38h1f821a2_0    conda-forge
openjpeg                  2.3.1                h254dc36_3    conda-forge
openssl                   1.1.1g               h0b31af3_0    conda-forge
osmnx                     0.14.1             pyh9f0ad1d_0    conda-forge
owslib                    0.20.0                     py_0    conda-forge
packaging                 20.4               pyh9f0ad1d_0    conda-forge
pandas                    1.0.4            py38h5fc983b_0    conda-forge
pandoc                    2.9.2.1                       0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
parso                     0.7.0              pyh9f0ad1d_0    conda-forge
pcre                      8.44                 h4a8c4bd_0    conda-forge
pexpect                   4.8.0            py38h32f6830_1    conda-forge
pickleshare               0.7.5           py38h32f6830_1001    conda-forge
pip                       20.1.1                     py_1    conda-forge
pixman                    0.38.0            h01d97ff_1003    conda-forge
plotly                    4.8.1              pyh9f0ad1d_0    conda-forge
ply                       3.11                       py_1    conda-forge
poppler                   0.87.0               h3232a60_1    conda-forge
poppler-data              0.4.9                         1    conda-forge
postgresql                12.2                 h62ab893_1    conda-forge
proj                      7.0.0                hbce2d5c_4    conda-forge
prometheus_client         0.8.0              pyh9f0ad1d_0    conda-forge
prompt-toolkit            3.0.5                      py_0    conda-forge
psycopg2                  2.8.5            py38hb761dab_1    conda-forge
ptyprocess                0.6.0                   py_1001    conda-forge
pycparser                 2.20                       py_0    conda-forge
pygments                  2.6.1                      py_0    conda-forge
pyopenssl                 19.1.0                     py_1    conda-forge
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
pyproj                    2.6.1.post1      py38h7d7e87d_0    conda-forge
pyqt                      5.12.3           py38hf180056_3    conda-forge
pyqt5-sip                 4.19.18                  pypi_0    pypi
pyqtchart                 5.12                     pypi_0    pypi
pyqtwebengine             5.12.1                   pypi_0    pypi
pyrsistent                0.16.0           py38h64e0658_0    conda-forge
pysocks                   1.7.1            py38h32f6830_1    conda-forge
python                    3.8.3           cpython_hd5f0129_0    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python_abi                3.8                      1_cp38    conda-forge
pytz                      2020.1             pyh9f0ad1d_0    conda-forge
pyyaml                    5.3.1            py38h64e0658_0    conda-forge
pyzmq                     19.0.1           py38h1fcdcd6_0    conda-forge
qca                       2.2.1                hc763db3_3    conda-forge
qgis                      3.12.2           py38h7226ff3_0    conda-forge
qjson                     0.9.0             hc763db3_1006    conda-forge
qscintilla2               2.11.2           py38h2815f34_2    conda-forge
qt                        5.12.5               h514805e_3    conda-forge
qtkeychain                0.10.0               h2815f34_0    conda-forge
qtlocation                5.12.5               h4a8c4bd_0    conda-forge
qtserialport              5.9.8                hc763db3_1    conda-forge
qwt                       6.1.5                h2815f34_0    conda-forge
qwtpolar                  1.1.1                h2815f34_7    conda-forge
readline                  8.0                  hcfe32e1_0    conda-forge
requests                  2.23.0             pyh8c360ce_2    conda-forge
retrying                  1.3.3                      py_2    conda-forge
rtree                     0.9.4            py38h08f867b_1    conda-forge
scikit-learn              0.23.1           py38hb7f0800_0    conda-forge
scipy                     1.4.1            py38h38b60c6_3    conda-forge
send2trash                1.5.0                      py_0    conda-forge
setuptools                47.1.1           py38h32f6830_0    conda-forge
shapely                   1.7.0            py38h5b1fbde_3    conda-forge
sip                       4.19.22          py38hc84c608_0    conda-forge
six                       1.15.0             pyh9f0ad1d_0    conda-forge
sqlite                    3.30.1               h93121df_0    conda-forge
tbb                       2019.9               ha1b3eb9_1    conda-forge
terminado                 0.8.3            py38h32f6830_1    conda-forge
testpath                  0.4.4                      py_0    conda-forge
threadpoolctl             2.1.0              pyh5ca1d4c_0    conda-forge
tiledb                    1.7.7                h84aa2a7_3    conda-forge
tk                        8.6.10               hbbe82c9_0    conda-forge
tornado                   6.0.4            py38h64e0658_1    conda-forge
traitlets                 4.3.3            py38h32f6830_1    conda-forge
tzcode                    2020a                h0b31af3_0    conda-forge
urllib3                   1.25.9                     py_0    conda-forge
wcwidth                   0.2.4              pyh9f0ad1d_0    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.34.2                     py_1    conda-forge
xerces-c                  3.2.2             h8f8adb3_1004    conda-forge
xz                        5.2.5                h0b31af3_0    conda-forge
yaml                      0.2.5                h0b31af3_0    conda-forge
zeromq                    4.3.2                h6de7cb9_2    conda-forge
zipp                      3.1.0                      py_0    conda-forge
zlib                      1.2.11            h0b31af3_1006    conda-forge
zstd                      1.4.4                h4b3e974_3    conda-forge
</details>

#### Code snippet that independently and completely reproduces the issue

```python
place = 'Kirkwood, Missouri, USA'
cf = '["highway"~"motorway|motorway_link|primary|primary_link|secondary_link|tertiary|' +\
        'tertiary_link|trunk|trunk_link|unclassified|secondary"]'

G_test = ox.graph_from_place(place, network_type='drive', custom_filter=cf)
G_test = ox.project_graph(G_test, to_crs='epsg:3857')
G_test = ox.consolidate_intersections(G_test, tolerance=20)

ox.io.save_graph_geopackage(G_test, 'data/Kirkwood_test.gpkg')
```

**Comment by gboeing** [comment]
Thanks. I believe the issue is the `u_original` and `v_original` attributes that `consolidate_intersections` adds to edges. These have a 64-bit int datatype, which Fiona doesn't like, and I need to add them to the list of attributes to convert to/from string on save/load. Should be an easy fix.

**Comment by gboeing** [comment]
Possibly related:
  - https://github.com/Toblerity/Fiona/issues/365
  - https://github.com/geopandas/geopandas/issues/348

**Comment by gboeing** [comment]
Actually, it looks like the problem is the consolidated graph's `u` and `v` edges attributes, which contain a mix of ints and strings in a column with `object` dtype.

**Comment by gboeing** [comment]
fixed in #517

**Comment by andrewalseth** [comment]
Thanks for addressing this quickly - I appreciate your development and maintenance of the package, it's incredibly useful
